### DISCUSSION TITLE
内存数据库不能共享多个虚拟机？

### DISCUSSION THREAD

**Discussion by terry8210** [starwing/lua-protobuf#170] [General]
内存数据库不能共享多个虚拟机？

**Comment by starwing** [comment]
看文档，`pb.state`，另外有个pr，你评估下哪种方案你比较喜欢~

**Reply by terry8210** [reply]
pb.state可以把当前内存数据库共享出来，然后设置给其他虚拟机使用？我想问是拷贝一份数据，还是共享一份内存指针？

**Reply by terry8210** [reply]
你说的pr是什么啊？

**Reply by starwing** [reply]
Pull Request， 这个：
![image](https://user-images.githubusercontent.com/114483/149759286-53849b1d-556c-43d9-9253-130983c46211.png)

这里是其他人提的我还没有合入的代码（主要是担心兼容性）

**Reply by terry8210** [reply]
好的，我看看

**Reply by terry8210** [reply]
pb.state 是拷贝一份数据，还是共享一份内存指针？

**Reply by starwing** [reply]
文档里写了啊，传一个nil会自动创建一个新的，你看看文档嘛……

**Reply by terry8210** [reply]
我看你最新代码中，有一个pb.unsafe.use的有一个接口，就是让vm自动选择使用本地还是全局的state

**Reply by starwing** [reply]
对的，这个可以在多个Lua VM里共享内存数据库，注意那个给出共享的那个Lua VM一定要最后释放。
